api/app/signals/apps/api/v1/views/signal.py

```diff
@@ -22,10 +21,20 @@
 from signals.auth.backend import JWTAuthBackend
 
 
-class PublicSignalViewSet(CreateModelMixin, DetailSerializerMixin, RetrieveModelMixin,
-                          PublicSignalGenericViewSet):
-    serializer_class = PublicSignalCreateSerializer
-    serializer_detail_class = PublicSignalSerializerDetail
+class PublicSignalViewSet(PublicSignalGenericViewSet):
+    def create(self, request):
+        serializer = PublicSignalCreateSerializer(data=request.data, context={'request': request})
+        serializer.is_valid(raise_exception=True)
+        signal = serializer.save()
+
+        data = PublicSignalSerializerDetail(signal, context={'request': request}).data
```

Comment:
Instead of `context={'request': request}` just use `context=self.get_serializer_context()`. This way you always pass the correct context to the serializer (how DRF want it to be)